__tests__/app.test.js

```diff
@@ -146,34 +140,81 @@ describe("GET /api/articles/:article_id/comments", () => {
+          expect(typeof comment.author).toBe("string");
+          expect(typeof comment.body).toBe("string");
+          expect(typeof comment.created_at).toBe("string");
         });
       });
   });
   test("GET:200 status responds with the most recent comments first.", () => {
     return request(app)
-    .get("/api/articles/9/comments")
-    .expect(200)
-    .then((res) => {
-      const comment = res.body.comments;
-      expect(comment).toBeSortedBy('created_at', {descending: true})
-
-    })
-  })
-  test("GET:204 responds with a message of no comments when the articles doesnt have any comments", () => {
+      .get("/api/articles/9/comments")
+      .expect(200)
+      .then((res) => {
+        const comment = res.body.comments;
+        expect(comment).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("GET:404 responds with a message of no comments when the articles doesnt have any comments", () => {
     return request(app)
-    .get("/api/articles/13/comments")
-    .expect(404)
-    .then((res) => {
-      const err = res.body.msg
+      .get("/api/articles/13/comments")
+      .expect(404)
+      .then((res) => {
+        const err = res.body.msg;
+        expect(err).toBe("no comments on this article");
+      });
+  });
+});
 
-      expect(err).toBe("no comments on this article");
-    })
-    })
-  })
+describe("POST: /api/articles/:article_id/comments ", () => {
+  test("POST: 201 status responds with the addition of the new comment in the database", () => {
+    const newComment = {
+      username: "lurker",
+      body: "Why do they call it rush hour when nothing moves?",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        const newComment = res.body.newComment;
+        expect(newComment.article_id).toBe(2);
+        expect(newComment.author).toBe("lurker");
+        expect(newComment.body).toBe(
+          "Why do they call it rush hour when nothing moves?"
+        );
+      });
+  });
+  test("POST: 404 error is returned when a username is given that does not exist", () => {
+    const newComment = {
+      username: "invalidUsername",
+      body: "Why do they call it rush hour when nothing moves?",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        const err = res.body.msg;
+        expect(err).toBe("this username does not exist");
+      });
+  });
+})
+  test("POST: 400 error is returned when one or all of the required fields are left null", () => {
+    const newComment = {
+      username: "lurker",
+      body: null,
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        const err = res.body.msg;
+        expect(err).toBe("Bad request");
+      });
```

Comment:
There should also be POST tests for
Status 201 - ignores unnecessary properties
Status 400 - invalid ID
Status 404 - non existent ID